ystia/yorc
# Feature/gh 83 hostpool placement

# Pull Request description

## Description of the change

### What I did

Add policy types for hostspool:
- yorc.policies.hostspool.Placement (abstract policy)
- yorc.policies.hostspool.BinPackingPlacement (default placement policy)
- yorc.policies.hostspool.RoundRobinPlacement

Add PlacementPolicy field to struct hostspool.Allocation struct
Add electHostFromCandidates method to hostspool manager to apply placement policies

See related A4C-yorc-provider PR: https://github.com/alien4cloud/alien4cloud-yorc-provider/pull/17
See A4C documentation : http://alien4cloud.github.io/#applying-hosts-pool-placement-policies
See CI tests: https://github.com/ystia/yorc/blob/feature/godog-ci/testdata/ci/features/Placement.feature
### How to verify it
Create a hosts pool with 2 shareable nodes
Create an app with a 2-instances compute and deploy it with RoundRobin placement then BinPacking Placement then no placement.

You can see the hostpool placement with:
`yorc hp list -l myHpLocation`

### Description for the changelog

## Applicable Issues
#83
